jcapi-systems.go

```diff
@@ -0,0 +1,334 @@
+}
+
+type JCSSHDParam struct {
+	Name  string `json:name`
+	Value string `json:value`
+}
+
+type JCNetworkInterface struct {
+	Name     string `json:name`
+	Internal bool   `json:internal`
+	Family   string `json:family`
+	Address  string `json:address`
+}
+
+func SystemsToString(systems []JCSystem) string {
+	returnVal := ""
+
+	for _, system := range systems {
+		returnVal += system.ToString()
+	}
+
+	return returnVal
+}
+
+func (jcsystem JCSystem) ToString() string {
+	//
+	// WARNING: Never output password via this method, it could be logged in clear text
+	//
+	//	returnVal := fmt.Sprintf("JCUSER: Id=[%s] - UserName=[%s] - FName/LName=[%s/%s] - Email=[%s] - sudo=[%t] - Uid=%s - Gid=%s - enableManagedUid=%t\n",
+	//		jcuser.Id, jcuser.UserName, jcuser.FirstName, jcuser.LastName,
+	//		jcuser.Email, jcuser.Sudo, jcuser.Uid, jcuser.Gid, jcuser.EnableManagedUid)
+	//
+	//	returnVal += fmt.Sprintf("JCUSER: ExternallyManaged=[%t] - ExternalDN=[%s] - ExternalSourceType=[%s]\n",
+	//		jcuser.ExternallyManaged, jcuser.ExternalDN, jcuser.ExternalSourceType)
+	//
+	//	for _, tag := range jcuser.Tags {
+	//		returnVal += fmt.Sprintf("\t%s\n", tag.ToString())
+	//	}
+	returnVal := "I'm a return val"
+	return returnVal
+}
+
+func getJCSSHDParamFieldsFromInterface(fields map[string]interface{}, params *JCSSHDParam) {
+	params.Name = fields["name"].(string)
+	params.Value = fields["value"].(string)
+}
+
+func getJCSSHDParamFromArray(paramArray []interface{}) []JCSSHDParam {
+	var returnVal []JCSSHDParam
+	returnVal = make([]JCSSHDParam, len(paramArray))
+
+	for idx, rec := range paramArray {
+		getJCSSHDParamFieldsFromInterface(rec.(map[string]interface{}), &returnVal[idx])
+	}
+
+	return returnVal
+
+}
+
+func getJCNetworkInterfaceFieldsFromInterface(fields map[string]interface{}, nic *JCNetworkInterface) {
+	nic.Address = fields["address"].(string)
+	nic.Family = fields["family"].(string)
+	nic.Internal = fields["internal"].(bool)
+	nic.Name = fields["name"].(string)
+}
+
+func getJCNetworkInterfacesFromArray(nicArray []interface{}) []JCNetworkInterface {
+	var returnVal []JCNetworkInterface
+	returnVal = make([]JCNetworkInterface, len(nicArray))
+
+	for idx, rec := range nicArray {
+		getJCNetworkInterfaceFieldsFromInterface(rec.(map[string]interface{}), &returnVal[idx])
+	}
+
+	return returnVal
+}
+
+func getJCSystemFieldsFromInterface(fields map[string]interface{}, system *JCSystem) {
+	// doing this b/c the jsonSysRec that's returned from the update only has a subset
+	// of the fields
+	if _, exists := fields["os"]; exists {
+		system.Os = fields["os"].(string)
+	}
+	if _, exists := fields["templateName"]; exists {
+		system.TemplateName = fields["templateName"].(string)
+	}
+	if _, exists := fields["allowSsgRootLogin"]; exists {
+		system.AllowSshRootLogin = fields["allowSshRootLogin"].(bool)
+	}
+	if _, exists := fields["id"]; exists {
+		system.Id = fields["id"].(string)
+	} else if _, exists := fields["_id"]; exists {
```

Comment:
heh yeah. my intention was to make sure that either id or _id gets put in the system.id field. i'll fix that.